### ISSUE TITLE
Auto Upload Jar

### ISSUE THREAD

**Issue by quasiben** [blaze/rambling#1]
rambling should be capable of bootstrapping itself similar to spark.  For example, spark apps build a directory like: `/user/ubuntu/.sparkStaging/application_1452033885404_0005` with the following contents:
- __spark_conf__3428001626499023137.zip
- py4j-0.8.2.1-src.zip
- pyspark.zip
- spark-assembly-1.5.2-hadoop2.6.0.jar

**Comment by mrocklin** [comment]
My previous understanding was that we assume that Yarn is widely deployed and that we only use rambling as a local client interface to Yarn.  What needs to be bootstrapped/deployed?

**Comment by quasiben** [comment]
I believe the JAR which launches the YARN job must also be inside of HDFS.  The ApplicationMaster is going to call on that JAR with similar arguments as the client.

For example:

## Client Call

On edge node/user's machine

```
com.continuumio.rambling.Client <args> 
```

## AppMaster Call

Inside ApplicationMaster Container

```
com.continuumio.rambling.ApplicationMaster <args> 
```

**Comment by quasiben** [comment]
Resolved by #13
